src/components/TodoItems.tsx

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import cn from 'classnames';
+import React, {
+  useContext,
+  useState,
+  useEffect,
+  useCallback,
+  useRef,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { TodoContext } from './TodoContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItems: React.FC<Props> = React.memo(({ todo }) => {
+  const { setTodos, todos, setChecked } = useContext(TodoContext);
+  const [title, setTitle] = useState(todo.title);
+  const [editing, setEditing] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const thisTodo = useRef<HTMLInputElement>(null);
+
+  const handleCheckedTodo = (todoId: number) => {
+    const newTodos = todos.map(curTodo => (
+      curTodo.id === todoId
+        ? { ...curTodo, completed: !curTodo.completed }
+        : curTodo));
+
+    setTodos(newTodos);
+    setChecked(newTodos.every(curTodo => curTodo.completed));
+  };
+
+  function deletedTodo(selectedTodo: Todo): void {
+    const filteredTodos = todos
+      .filter(t => t !== selectedTodo);
+
+    setTodos([...filteredTodos]);
+  }
+
+  const handleDoubleClick = useCallback((
+    event: React.MouseEvent<HTMLLabelElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    setEditing(true);
+    setFocus(true);
+  }, []);
+
+  useEffect(() => {
+    if (focus && thisTodo.current) {
+      thisTodo.current.focus();
+    }
+  }, [focus]);
+
+  const handleBlur = () => {
+    setEditing(false);
+    setFocus(false);
+
+    if (title.trim() === '') {
+      deletedTodo(todo);
+
+      return;
+    }
+
+    setTodos(todos.map(t => (t !== todo
```

Comment:
Don't cut variable names, try to rewrite with more meaningful variable naming 
```suggestion
    setTodos(todos.map(t => (t !== todo
```